deprish_to_eia.py

```diff
@@ -0,0 +1,287 @@
+# Prep the inputs
+###############################################################################
+
+
+def prep_int_ids(int_ids):
+    """Prep dictionary of column names (key) with nullable int dype (value)."""
+    # prep dtype arg for integer columns for read_excel
+    dtypes_dict = {}
+    for i in int_ids:
+        dtypes_dict[i] = pd.Int64Dtype()
+    return dtypes_dict
+
+
+def grab_sheet_from_excel(file_path, sheet_name, dtypes_dict):
+    """Grab a sheet from an excel file."""
+    logger.info(f'grabbing sheet from: {file_path}')
+    deprish_df = pd.read_excel(
+        file_path, skiprows=1, sheet_name=sheet_name, dtype=dtypes_dict)
+    return deprish_df
+
+
+def clean_strings(df, key, strings_to_clean):
+    """Clean the strings in a given column."""
+    df = pudl.helpers.strip_lower(df, [key])
+    df.loc[:, key] = df.loc[:, key].replace(strings_to_clean, regex=True)
+    return df
+
+
+def prep_deprish(file_path_deprish, sheet_name_deprish, plant_parts_df, key1):
+    """Prep the depreciation dataframe."""
+    logger.info(f"Reading the depreciation data from {file_path_deprish}")
+    deprish_df = (pd.read_excel(
+        file_path_deprish, skiprows=0, sheet_name=sheet_name_deprish,
+        dtypes=prep_int_ids(INT_IDS))
+        .astype({'report_date': 'datetime64[ns]',
+                 'plant_id_pudl': pd.Int64Dtype()}))
+    ids = ['plant_id_pudl', 'report_year']
+    deprish_ids = (pd.merge(
+        (deprish_df
+         .astype({'report_date': 'datetime64[ns]'})
+         .assign(report_year=deprish_df.report_date.dt.year)
+         .dropna(subset=ids)
+         ),
+        plant_parts_df[ids].drop_duplicates(),
+        how='outer', indicator=True, on=ids))
+    deprish_ids = (deprish_ids.loc[deprish_ids._merge == 'both']
+                   .drop_duplicates(subset=['plant_name'])
+                   .dropna(subset=['plant_name'])
+                   )
+    return clean_strings(deprish_ids, key1, STRINGS_TO_CLEAN)
+
+
+def prep_master_parts_eia(plant_parts_df, key2):
+    """Prepare the EIA master plant parts."""
+    plant_parts_df = (clean_strings(plant_parts_df, key2, STRINGS_TO_CLEAN)
+                      .reset_index())
+    return plant_parts_df
+
+
+###############################################################################
+# Fuzzy Matching
+###############################################################################
+
+
+def get_plant_year_list(plant_name, df1, df2, key1, key2):
+    """
+    Get the possible key matches from df2 a plant_id_pudl and report_year.
+
+    This selects for the plant id and year for each of the df1 records to
+    match. This is for use within `get_fuzzy_matches`.
+    """
+    logger.debug(plant_name)
+    plant_id_pudl = df1.loc[df1.plant_name ==
+                            plant_name, 'plant_id_pudl'].values[0]
+    report_year = df1.loc[df1.plant_name ==
+                          plant_name, 'report_year'].values[0]
```

Comment:
This is perhaps me just not knowing the correct syntax to return just the string contents of a cell in a dataframe. df1 is the depreciation sheet and there is only one entry per name/year. I suppose I could make this more general and grab all of the report_years and plant_id_pudls and make 
```
names = df2.loc[(df2.plant_id_pudl.isin(plant_id_pudls)) & (df2.report_year.isin( report_years))][key2].to_list()

Reply:
Another option might be to use [`pandas.DataFrame.at`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.at.html) which works like `df.loc[]` but is specifically for pulling the value at a single particular combination of indices. I think it would also fail if you unexpectedly had more than one record returned by the index lookup, which might be desirable.